File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/bridge/uielement/Table.java
package com.bigboxcode.designpattern.bridge.uielement;

public class Table extends UIElement{

    public Table(ColorSchema colorSchema) {
        super(colorSchema);
    }

    @Override
    public void printElement() {
        // Write code for printing table element
        System.out.println("Printing Table");

        // Set color schema
        colorSchema.setColor();
    }
}

File: design-patterns/ts/bridge/ui-element/table.ts
// table.ts

import Color from "./color";
import UIElement from "./ui-element";

class Table extends UIElement {
    constructor(color: Color) {
        super(color);
    }
    public printElement(): void {
        this.color.setColor();

        console.log("Printing Table");
    }
}

export default Table;